### PR TITLE
skytemple-files: add support for patches

### PR DESCRIPTION
**Pull request by marius851000**
###### Motivation for this change
I wanted to use the patch functionallity, but it appeared to not work.

###### Things done
Packaged armips (used to apply patch)
made a patch for skytemple-files (merged upstream) to ensure that file copied from the nix store to the temporary directory is writable by the program.
Added myself as a maintainers of the thing I added.

tested the application of some patches

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@xfix

### CODE REVIEW

File: pkgs/development/python-modules/skytemple-files/default.nix [hunk 1]
```diff
@@ -8,9 +8,26 @@ buildPythonPackage rec {
...
     rev = version;
-    sha256 = "1vklg4kcj3kb9ryrzrcmywn131b2bp3vy94cd4x4y4s7hkhgwg74";
+    sha256 = "sha256-/S0otBujwO/IMiLKgA2o8wlD6xk1/DpwOAfemojV9NU=";
+    fetchSubmodules = true;
   };
 
+  patches = [
+    # fix patching https://github.com/SkyTemple/skytemple-files/pull/128
+    # merged, remove for next update
+    (fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/skytemple-files/default.nix:18]: You must use fetchpatch here and update the has.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126676` at 8b364280 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built successfully:</summary>
  <ul>
    <li>python38Packages.skytemple-dtef</li>
    <li>python38Packages.skytemple-files</li>
    <li>python38Packages.skytemple-ssb-debugger</li>
    <li>python39Packages.skytemple-dtef</li>
    <li>python39Packages.skytemple-files</li>
    <li>python39Packages.skytemple-ssb-debugger</li>
    <li>skytemple</li>
  </ul>
</details>

File: pkgs/development/compilers/armips/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "Kingcom";
+    repo = "armips";
+    rev = "v${version}";
+    sha256 = "sha256-L+Uxww/WtvDJn1xZqoqA6Pkzq/98sy1qTxZbv6eEjbA=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  installPhase = ''
```
Comment (xfix) [inline pkgs/development/compilers/armips/default.nix:16]: Please run install hooks, see https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md.

**Review by sugar700 — CHANGES_REQUESTED** [review]
Tested patches funcitonality, it works after this change.

**Review by sugar700 — APPROVED** [review]

**Pull request merged**